lib/storage/metadata/oplog/OplogInterface.js

```diff
@@ -0,0 +1,147 @@
+            this.persist = params.persist;
+        }
+        if (params && params.persistData !== undefined) {
+            this.persistData = params.persistData;
+        }
+        this.logger = new werelogs.Logger('OplogInterface');
+        /* for backends requiring bufferization only */
+        this.bufferTimeoutMs = 500;
+        if (params && params.bufferTimeoutMs !== undefined) {
+            this.bufferTimeoutMs = params.bufferTimeoutMs;
+        }
+        this.addQueue = [];
+        this.delQueue = [];
+        this.pending = false;
+        this.prevOffset = null;
+        this.offset = null;
+    }
+
+    addEvent(item, offset) {
+        this.addQueue.push(item);
+        this.offset = offset;
+    }
+
+    delEvent(item, offset) {
+        this.delQueue.push(item);
+        this.offset = offset;
+    }
+
+    /*
+     * Optional buffer management for backends that don't bufferize.
+     * It avoids persisting the state at each event
+     */
+    flushQueue(cb) {
+        // console.log(this.prevOffset, this.offset);
+        if (this.offset === null ||
+            this.prevOffset === this.offset) {
+            if (cb) {
+                return process.nextTick(cb);
+            }
+            return undefined;
+        }
+        if (this.pending) {
+            if (cb) {
+                return process.nextTick(cb);
+            }
+            return undefined;
+        }
+        this.pending = true;
+        const addQueue = this.addQueue;
+        this.addQueue = [];
+        const delQueue = this.delQueue;
+        this.delQueue = [];
+        const offset = this.offset;
+        this.prevOffset = this.offset;
+        this.persistData.updateState(
+            addQueue,
+            delQueue,
+            err => {
+                if (err) {
+                    if (cb) {
+                        return cb(err);
+                    }
+                }
+                this.persist.save(
+                    this.filterName,
+                    this.persistData,
+                    offset,
+                    err => {
+                        this.pending = false;
+                        if (err) {
+                            if (cb) {
+                                return cb(err);
+                            }
+                            return undefined;
+                        }
+                        if (cb) {
+                            return cb();
+                        }
+                        return undefined;
+                    });
+                return undefined;
+            });
+        return undefined;
+    }
+
+    doFlush() {
+        this.flushQueue(err => {
+            if (err) {
+                this.logger.error('flusing buffer', { err });
+            }
+        });
+        this.startFlusher();
+    }
+
+    startFlusher() {
+        setTimeout(this.doFlush.bind(this), this.bufferTimeoutMs);
+    }
+
+    // method to be overriden
```

Comment:
```suggestion
    // method to be overridden
```

Reply:
If this method is always meant to be overridden, it should throw `NotImplemented` instead, to prevent silent mistakes from happening.